common/components/Header/components/CustomNodeModal.tsx

```diff
@@ -0,0 +1,230 @@
+                {this.renderInput({
+                  name: 'port',
+                  placeholder: '8545',
+                  type: 'number',
+                }, invalids)}
+              </div>
+            </div>
+            <div className="row">
+              <div className="col-sm-12">
+                <label>
+                  <input
+                    type="checkbox"
+                    name="hasAuth"
+                    checked={this.state.hasAuth}
+                    onChange={this.handleCheckbox}
+                  />
+                  {' '}
+                  <span>HTTP Basic Authentication</span>
+                </label>
+              </div>
+            </div>
+            {this.state.hasAuth &&
+              <div className="row">
+                <div className="col-sm-6">
+                  <label>Username</label>
+                  {this.renderInput({ name: 'username' }, invalids)}
+                </div>
+                <div className="col-sm-6">
+                  <label>Password</label>
+                  {this.renderInput({
+                    name: 'password',
+                    type: 'password',
+                  }, invalids)}
+                </div>
+              </div>
+            }
+          </form>
+        </div>
+      </Modal>
+    );
+  }
+
+  private renderInput(input: Input, invalids: { [key: string]: boolean }) {
+    return <input
+      className={classnames({
+        'form-control': true,
+        'is-invalid': this.state[input.name] && invalids[input.name],
+      })}
+      value={this.state[name]}
+      onChange={this.handleChange}
+      {...input}
+    />;
+  }
+
+  private getInvalids(): { [key: string]: boolean } {
+    const {
+      url,
+      port,
+      hasAuth,
+      username,
+      password,
+    } = this.state;
+    const required = ["name", "url", "port", "network"];
+    const invalids: { [key: string]: boolean } = {};
+
+    // Required fields
+    required.forEach((field) => {
+      if (!this.state[field]) {
+        invalids[field] = true;
+      }
+    });
+
+    // Somewhat valid URL, not 100% fool-proof
+    if (!/https?\:\/\/\w+/i.test(url)) {
+      invalids.url = true;
+    }
+
+    // Numeric port within range
+    const iport = parseInt(port, 10);
+    if (!iport || iport < 1 || iport > 65535) {
+      invalids.port = true;
+    }
+
+    // If they have auth, make sure it's provided
+    if (hasAuth) {
+      if (!username) {
+        invalids.username = true;
+      }
+      if (!password) {
+        invalids.password = true;
+      }
+    }
+
+    return invalids;
+  }
+
+  private handleChange = (ev: React.FormEvent<
+    HTMLInputElement | HTMLSelectElement
+  >) => {
+    const { name, value } = ev.currentTarget;
+    this.setState({ [name]: value } as any);
+  };
+
+  private handleCheckbox = (ev: React.FormEvent<HTMLInputElement>) => {
+    const { name } = ev.currentTarget;
+    this.setState({ [name]: !this.state[name] } as any);
```

Comment:
@HenryNguyen5 can you check these two `as any`s out?

Reply:
Checked these out, looks like a legit bug.
One thing you can do is move `as any` inline so it looks like `this.setState({ [name as any]: !this.state[name] }) so you're only setting the computed property name as `any` and not the entire object itself.